### PR TITLE
Refactor signature-patching function to handle search and distinct for FastAPI

### PR DESCRIPTION
**Pull request by jmaruland**
In Progress. Created one method that helps FastAPI generate documentation for search() and distinct. Currently, the server is registering the methods but it's not finding any of them when they are called from the client side

### CODE REVIEW

**Comment by jmaruland** [comment]
Adding `distinct()` at the end of a sequence of search queries, like
`client.search(query1).search(query2)....distinct(key)`
applies the `distinct` request to the search result. Before, the `distinct()` request was being run from the perspective of the root node. This last commit gives more freedom to the use of `distinct()`

File: tiled/server/app.py [hunk 1]
```diff
@@ -324,6 +326,21 @@ async def index(
         # And add this authentication_router itself to the app.
         app.include_router(authentication_router, prefix="/api/v1/auth")
 
+    # The /search route is defined at server startup so that the user has the
```
Comment (danielballan) [inline tiled/server/app.py:329]:
```suggestion
    # The /search route is defined after import time so that the user has the
```

**Review by danielballan — COMMENTED** [review]
Two minor comments to keep comments up to date and avoid cruft.

**Review by danielballan — APPROVED** [review]

**Comment by danielballan** [comment]
I took this for a test drive locally. Works as advertised.

**Pull request merged**